src/Util/FormViewIterator.php

```diff
@@ -29,7 +29,10 @@ class FormViewIterator implements \RecursiveIterator
 
     public function __construct(FormView $formView)
     {
-        $this->iterator = $formView->getIterator();
+        $iterator = $formView->getIterator();
+        // todo - remove `assert` statement after https://github.com/symfony/symfony/pull/37760 is merged
+        \assert($iterator instanceof \ArrayIterator);
```

Comment:
I just found this
https://github.com/phpstan/phpstan/issues/2494

And made a try
https://phpstan.org/r/80c83e17-7c54-4e17-8111-d52dc657077c

What was the phpstan error ?

Reply:
By the way, do we use phpstan/phpstan-symfony ?

It provides a stub https://github.com/phpstan/phpstan-symfony/blob/master/stubs/FormView.stub

Reply:
Ok, I found the error, I think you can close the symfony PR @vladyslavstartsev.

It's seems related to `phpstan/phpstan-symfony` instead.
https://github.com/phpstan/phpstan-symfony/blob/master/stubs/FormView.stub

If I remove the stub in my vendor, there is no errors.

Reply:
Seems like we can avoid using `\assert` by simply adding precision in the property typehint.

```
/**
     * @var \ArrayIterator<string, FormView>
     */
    protected $iterator;
```

Reply:
or <int, FormView>

Reply:
I don't know if it's right, but the stub implement `IteratorAggregate<string, self>`.
So it has to be \ArrayIterator<string, FormView>